gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/kmeans/KMeans.java

```diff
@@ -0,0 +1,150 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.impl.operators.kmeans;
+
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.functions.KeySelector;
+import org.apache.flink.api.java.operators.IterativeDataSet;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.utils.DataSetUtils;
+import org.gradoop.common.model.api.entities.Edge;
+import org.gradoop.common.model.api.entities.GraphHead;
+import org.gradoop.common.model.api.entities.Vertex;
+import org.gradoop.flink.model.api.epgm.BaseGraph;
+import org.gradoop.flink.model.api.epgm.BaseGraphCollection;
+import org.gradoop.flink.model.api.operators.UnaryBaseGraphToBaseGraphOperator;
+import org.gradoop.flink.model.impl.operators.kmeans.functions.SelectNearestCenter;
+import org.gradoop.flink.model.impl.operators.kmeans.functions.VertexPostProcessingMap;
+import org.gradoop.flink.model.impl.operators.kmeans.functions.CountAppender;
+import org.gradoop.flink.model.impl.operators.kmeans.functions.CentroidAccumulator;
+import org.gradoop.flink.model.impl.operators.kmeans.functions.CentroidAverager;
+import org.gradoop.flink.model.impl.operators.kmeans.functions.PointToKey;
+import org.gradoop.flink.model.impl.operators.kmeans.util.Centroid;
+import org.gradoop.flink.model.impl.operators.kmeans.util.Point;
+
+/**
+ * Takes a logical graph, an user defined amount of iterations and centroids, and the property names
+ * of the vertex that are used for the clustering as input. Adds the clusterId, together with the
+ * cluster coordinates to the properties of the vertex. Returns the logical graph with modified
+ * vertex properties.
+ *
+ * @param <G>  The graph head type.
+ * @param <V>  The vertex type.
+ * @param <E>  The edge type.
+ * @param <LG> The type of the graph.
+ * @param <GC> The type of the graph collection.
+ */
+
+
+public class KMeans<G extends GraphHead, V extends Vertex, E extends Edge, LG extends BaseGraph<G, V, E, LG, GC>, GC extends BaseGraphCollection<G, V, E, LG, GC>> implements
+  UnaryBaseGraphToBaseGraphOperator<LG> {
+
+  /**
+   * Amount if times the algorithm iterates over the vertices
+   */
+  private final int iterations;
+
+  /**
+   * Amount of clusters
+   */
+  private final int centroids;
+
+  /**
+   * First spatial property used for the clustering
+   */
+  private final String latName;
+
+  /**
+   * Second spatial property used for the clustering
+   */
+  private final String lonName;
+
+  /**
+   * Constructor to create an instance of KMeans
+   *
+   * @param iterations        Amount of times the algorithm iterates over the the vertices
```

Comment:
```suggestion
   * @param iterations        Number of iterations, e.g., xx
```
What do you think is a good number of iterations? 20 ?

Reply:
20 is fine. Changed comment, used 20 as an example